app/models/agents/http_status_agent.rb

```diff
@@ -47,23 +49,30 @@ def validate_options
     end
 
     def check
-      check_this_url interpolated[:url]
+      check_this_url interpolated[:url], interpolated[:header]
     end
 
     def receive(incoming_events)
       incoming_events.each do |event|
         interpolate_with(event) do
-          check_this_url interpolated[:url]
+          check_this_url interpolated[:url], interpolated[:header]
         end
       end
     end
 
     private
 
-    def check_this_url(url)
+    def check_this_url(url, header)
+      # Track time
       measured_result = TimeTracker.track { ping(url) }
+      # Deal with failures
       if measured_result.result
-        create_event payload: { 'url' => url, 'status' => measured_result.status.to_s, 'response_received' => true, 'elapsed_time' => measured_result.elapsed_time }
+        # Deal with headers
+        if !header.empty?
```

Comment:
I think the array support would be nice, and calling the option `headers`.  It could probably work like this?

```ruby
payload = { 'url' => url, 'status' => measured_result.status.to_s, 'response_received' => true, 'elapsed_time' => measured_result.elapsed_time }

if headers.present?
  payload.merge!(headers: measured_result.result.headers.slice(*Array(headers)))
end

create_event payload: payload
```

That's a bit confusing, but it should accept either `'Server'` or `['Server', 'Content-Type]`, and return a nested hash with only those selected headers.